library/src/main/java/com/firebase/ui/auth/core/FirebaseLoginBaseActivity.java

```diff
@@ -0,0 +1,109 @@
+package com.firebase.ui.auth.core;
+
+import android.content.Intent;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+
+import com.firebase.client.AuthData;
+import com.firebase.client.Firebase;
+import com.firebase.client.FirebaseError;
+import com.firebase.ui.auth.facebook.FacebookAuthHelper;
+import com.firebase.ui.auth.google.GoogleAuthHelper;
+import com.firebase.ui.auth.password.PasswordAuthHelper;
+import com.firebase.ui.auth.twitter.TwitterAuthHelper;
+
+public abstract class FirebaseLoginBaseActivity extends AppCompatActivity {
+
+    private final String TAG = "FirebaseLoginBaseAct";
+
+    private Firebase.AuthStateListener mAuthStateListener;
+    private FirebaseLoginDialog mDialog;
+    private TokenAuthHandler mHandler;
+
+    /* Abstract methods for Login Events */
+    protected abstract void onFirebaseLoggedIn(AuthData authData);
```

Comment:
Add JavaDoc (with sample code snippets) of what devs should do in these overridden methods.